expan/core/experiment.py

```diff
@@ -218,26 +216,33 @@ def outlier_filter(self, data, kpis, percentile=99.0, threshold_type='upper'):
                         "Denominator '{}' of the derived KPI does not exist in the data in outlier filtering.".format(kpi.denominator))
                 kpi.make_derived_kpi(data)
 
-        # check if provided percentile is valid
-        if 0.0 < percentile <= 100.0 is False:
-            raise ValueError("Percentile value needs to be between 0.0 and 100.0!")
-        # check if provided filtering kind is valid
-        if threshold_type not in ['upper', 'lower']:
-            raise ValueError("Threshold type needs to be either 'upper' or 'lower'!")
+        admittable_thresholds = set(['upper', 'lower', 'two-sided',
+                                     'two-sided-asym'])
+
+        thresholds = thresholds or {}
+        for kpi in thresholds.keys():
+            threshold_type, percentile = thresholds[kpi]
+
```

Comment:
`for kpi, (threshold_type, percentile) in thresholds.items():` is another possibility

(I thought I made this comment already, but I can't find it. Sorry for any repetition, and sorry if I missed any response)

Reply:
You have suggested `for kpi in thresholds.keys():` before, indeed.

Reply:
Doh! I thought of both, but maybe I only mentioned one of them. 

Anyway, I'll resolve this now